CONTRIBUTING.md

```diff
@@ -0,0 +1,165 @@
+
+If you are a member of the `dbt-labs` GitHub organization, you will have push access to the `dbt` repo. Rather than forking `dbt-bigquery` to make your changes, just clone the repository, check out a new branch, and push directly to that branch.
+
+## Setting up an environment
+There are some tools that will be helpful to you in developing locally. While this is the list reveleant for `dbt` development, many of these tools are used commonly across open-source python projects.
+
+### Tools
+
+A short list of tools used in `dbt` testing that will be helpful to your understanding:
+
+- [`tox`](https://tox.readthedocs.io/en/latest/) to manage virtualenvs across python versions. We currently target the latest patch releases for Python 3.6, Python 3.7, Python 3.8, and Python 3.9
+- [`pytest`](https://docs.pytest.org/en/latest/) to discover/run tests
+- [`make`](https://users.cs.duke.edu/~ola/courses/programming/Makefiles/Makefiles.html) - but don't worry too much, nobody _really_ understands how make works and our Makefile is super simple
+- [`flake8`](https://flake8.pycqa.org/en/latest/) for code linting
+- [`mypy`](https://mypy.readthedocs.io/en/stable/) for static type checking
+- [Github Actions](https://github.com/features/actions)
+
+A deep understanding of these tools in not required to effectively contribute to `dbt`, but we recommend checking out the attached documentation if you're interested in learning more about them.
+
+#### virtual environments
+
+We strongly recommend using virtual environments when developing code in `dbt`. We recommend creating this virtualenv
+in the root of the `dbt` repository. To create a new virtualenv, run:
+```sh
+python3 -m venv env
+source env/bin/activate
+```
+
+This will create and activate a new Python virtual environment.
+
+#### docker and docker-compose
+
+Docker and docker-compose are both used in testing. Specific instructions for you OS can be found [here](https://docs.docker.com/get-docker/).
+
+## Running `dbt-bigquery` in development
+
+### Installation
+
+First make sure that you set up your `virtualenv` as described in [Setting up an environment](#setting-up-an-environment).  Also ensure you have the latest version of pip installed with `pip install --upgrade pip`. Next, install `dbt-biguery` latest dependencies:
+
+```sh
+pip install -r dev-requirements.txt
+```
+
+When `dbt` is installed this way, any changes you make to the `dbt` source code will be reflected immediately in your next `dbt` run.
+
+## Testing
+
+### Test commands
+There are a few mehtods for running tests locally.
+
+####`tox`
+`tox` takes care of managing virtualenvs and install dependencies in order to run tests. You can also run tests in parallel, for example you can run unit tests for Python 3.6,Python 3.7, Python 3.8, `flake8` checks and `mypy` checks in parallel with `tox -p`. Also you  can run unit tests for specific python versions with `tox -e py36`. The configuration of theses tests are located in `tox.ini`.
```

Comment:
- remove mention of py36 (true across the board)
- we don't currently run `mypy` checks on adapter plugins (note: **this is a place where adapter plugins differ from dbt-core**)